### PR TITLE
Update metrics-instrumentation.md

### PR DESCRIPTION
**Pull request by JamesNK**
Add example names to instrumentation docs to make the guidelines more understandable.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/diagnostics/metrics-instrumentation.md](https://github.com/dotnet/docs/blob/2be9f0bfbdc289664424d14bbb0be17c19b2d91c/docs/core/diagnostics/metrics-instrumentation.md) | [Creating Metrics](https://review.learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics-instrumentation?branch=pr-en-us-37060) |

### CODE REVIEW

File: docs/core/diagnostics/metrics-instrumentation.md [hunk 1]
```diff
@@ -83,10 +83,16 @@ which use dotted hierarchical names. Assembly names or namespace names for code
 for code in a second, independent assembly, the name should be based on the assembly that defines the Meter, not the assembly whose code is being instrumented.
 
 - .NET doesn't enforce any naming scheme for Instruments, but we recommend following
-[OpenTelemetry naming guidelines](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#general-guidelines), which use dotted hierarchical names
+[OpenTelemetry naming guidelines](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#general-guidelines), which use lowercase dotted hierarchical names
 and an underscore ('_') as the separator between multiple words in the same element. Not all metric tools preserve the Meter name as part of the final metric name, so it's beneficial
 to make the instrument name globally unique on its own.
 
+  Example instruments names:
```
Comment (JamesNK) [inline docs/core/diagnostics/metrics-instrumentation.md:90]:
```suggestion
  Example instrument names:
```

**Review by noahfalk — APPROVED** [review]
LGTM, thanks!

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

Let's :shipit:

File: docs/core/diagnostics/metrics-instrumentation.md [hunk 2]
```diff
@@ -490,19 +496,26 @@ Press p to pause, r to resume, q to quit.
 
 ### Best practices
 
-- Although the API allows any object to be used as the tag value, numeric types and strings are anticipated by collection tools. Other types may or may not be
+- Although the API allows any object to be used as the tag value, numeric types, and strings are anticipated by collection tools. Other types may or may not be
```
Comment (gewarren) [inline docs/core/diagnostics/metrics-instrumentation.md:499]:
```suggestion
- Although the API allows any object to be used as the tag value, numeric types and strings are anticipated by collection tools. Other types may or may not be
```

Comment (gewarren) [reply docs/core/diagnostics/metrics-instrumentation.md:499]: I think only numeric types and strings go together here, not all three.

**Review by gewarren — APPROVED** [review]

**Pull request merged**